core/src/main/java/com/google/zxing/datamatrix/decoder/DecodedBitStreamParser.java

```diff
@@ -554,5 +587,61 @@ private static int unrandomize255State(int randomizedBase256Codeword,
     int tempVariable = randomizedBase256Codeword - pseudoRandomNumber;
     return tempVariable >= 0 ? tempVariable : tempVariable + 256;
   }
-
+  private static final class ECIStringBuilder {
```

Comment:
>I don't quite follow this - you're recording chars as bytes, then converting back to chars. For ASCII, when would that yield any difference?
It would not yield any difference. I thought to use an array of bytes instead of chars when bytes are sufficient. I also makes it clear that everything is really in byte semantics and not character semantic as one might think.